masonry/src/widgets/flex.rs

```diff
@@ -736,29 +723,19 @@ impl Widget for Flex {
             }
         }
 
-        // figure out if we have extra space on major axis, and if so how to use it
-        let extra = if self.fill_major_axis {
-            (remaining - major_flex).max(0.0)
+        // COMPUTE EXTRA SPACE
+        let extra_length = if self.fill_major_axis {
+            (remaining_major - major_flex).max(0.0)
         } else {
-            // if we are *not* expected to fill our available space this usually
+            // If we are *not* expected to fill our available space this usually
             // means we don't have any extra, unless dictated by our constraints.
             (self.direction.major(bc.min()) - (major_non_flex + major_flex)).max(0.0)
         };
-
-        // the actual size needed to tightly fit the children on the minor axis.
-        // Unlike the 'minor' var, this ignores the incoming constraints.
-        let minor_dim = match self.direction {
-            Axis::Horizontal if any_use_baseline => max_below_baseline + max_above_baseline,
-            _ => minor,
-        };
```

Comment:
Remove the distinction between `minor` and `minor_dim`.

This distinction was only deliberately used in the `CrossAxisAlignment::Baseline` block below, where I moved this logic. Doing so lets us remove a bunch of code, including the `any_use_baseline` variable.

As a side-effect, this means that `minor` is used instead of `minor_dim` for the non-baseline blocks below, which I'm 99% sure will remove some corner-case jank (see further comments).